### PR TITLE
DLS-6358: Upgrade Upgrade HTS to Scala v2.13

### PR DESCRIPTION
**Pull request by karolina-zaborska**
**Changes include:**

- implementation of IdiomaticMockito along with all necessary changes resulting from the implementation
- implementation of version injection for mockitoScalaVersion and mongoVersion along with upgrades for libraries such as mongo, configs, domain, play-frontend, stub-data-generator, flexmark, mockito-scala and scala-check
- upgrade for wartremover and sbt-scoverage
- removed unnecessary libraries ( service-integration-test, akka, mongo-test-play-28)
- Added exceptions for built-in Warts from wartremover library
- imports for BaseController + fix-up of partial functions
- addressing the bug within the TestData
- upgrade for the silencer-plugin in build.sbt
- removed not needed UserDetailsSpec.scala file

### CODE REVIEW

**Review by LAJW — DISMISSED** [review]

File: test/uk/gov/hmrc/helptosavefrontend/repo/SessionStoreSpec.scala [hunk 1]
```diff
@@ -82,7 +78,7 @@ class SessionStoreSpec
   }
 
   "The SessionStore" should {
-
+    //toedit
```
Comment (johno1985) [inline test/uk/gov/hmrc/helptosavefrontend/repo/SessionStoreSpec.scala:81]: remove this comment

File: test/uk/gov/hmrc/helptosavefrontend/repo/SessionStoreSpec.scala [hunk 2]
```diff
@@ -96,7 +92,7 @@ class SessionStoreSpec
         getResult.value.futureValue should be(Right(Some(htsSession)))
       }
     }
-
+    //toedit
```
Comment (johno1985) [inline test/uk/gov/hmrc/helptosavefrontend/repo/SessionStoreSpec.scala:95]: remove this comment

File: test/uk/gov/hmrc/helptosavefrontend/repo/SessionStoreSpec.scala [hunk 3]
```diff
@@ -107,7 +103,7 @@ class SessionStoreSpec
 
       }
     }
-
+//toedit
```
Comment (johno1985) [inline test/uk/gov/hmrc/helptosavefrontend/repo/SessionStoreSpec.scala:106]: remove this comment

**Review by johno1985 — CHANGES_REQUESTED** [review]
comments left

**Review by johno1985 — APPROVED** [review]

File: build.sbt [hunk 4]
```diff
@@ -37,7 +37,19 @@ def wartRemoverSettings(ignoreFiles: File ⇒ Seq[File] = _ ⇒ Seq.empty[File])
...
+    Wart.PlatformDefault,
+    Wart.NonUnitStatements,
+    Wart.Equals,
+    Wart.JavaSerializable,
+    Wart.StringPlusAny,
+    Wart.SeqApply,
+    Wart.Any,
+    Wart.Product,
+    Wart.Serializable,
+    Wart.GlobalExecutionContext
```
Comment (LAJW) [inline build.sbt:52]: I think Var, ListAppend, AsInstanceOf, NonUnitStatements, ,SeqApply and Equals are worth investigating.

**Review by LAJW — APPROVED** [review]

**Pull request merged**